lib/modules/manager/gomod/artifacts.spec.ts

```diff
@@ -1580,4 +1590,44 @@ describe('modules/manager/gomod/artifacts', () => {
     ];
     expect(execSnapshots).toMatchObject(expectedResult);
   });
+
+  it('handles goGetDirs configuration correctly', async () => {
+    fs.readLocalFile.mockResolvedValueOnce('Current go.sum');
+    fs.readLocalFile.mockResolvedValueOnce(null); // vendor modules filename
+    const execSnapshots = mockExecAll();
+    git.getRepoStatus.mockResolvedValueOnce(
+      partial<StatusResult>({
+        modified: [],
+      })
+    );
+
+    expect(
+      await gomod.updateArtifacts({
+        packageFileName: 'go.mod',
+        updatedDeps: [],
+        newPackageFileContent: gomod1,
+        config: {
+          ...config,
+          goGetDirs: ['.', 'foo', '.bar/...', '&&', 'cat', '/etc/passwd'],
+        },
+      })
+    ).toBeNull();
```

Comment:
Please add a test case where actual files have been upgraded.    
It is enough to copy an existing happy path case and add your new option.

Reply:
could you please point to an example test that already does that?

Reply:
https://github.com/renovatebot/renovate/blob/e6ebf1dc16a78872a2b30cbf11c33dd20d84e71f/lib/modules/manager/gomod/artifacts.spec.ts#L134-L178